graphdatascience/graph/graph_entity_ops_runner.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Callable, Dict, List, Union
+
+from pandas.core.frame import DataFrame
+from pandas.core.series import Series
+
+from .graph_object import Graph
+from graphdatascience.caller_base import CallerBase
+from graphdatascience.error.illegal_attr_checker import IllegalAttrChecker
+from graphdatascience.error.uncallable_namespace import UncallableNamespace
+from graphdatascience.query_runner.query_runner import QueryRunner
+from graphdatascience.server_version.compatible_with import compatible_with
+from graphdatascience.server_version.server_version import ServerVersion
+
+Strings = Union[str, List[str]]
+
+
+class GraphEntityOpsBaseRunner(CallerBase, UncallableNamespace, IllegalAttrChecker):
+    def __init__(
+        self,
+        query_runner: QueryRunner,
+        namespace: str,
+        server_version: ServerVersion,
+        handle_properties: Callable[[Graph, Strings, Strings, Dict[str, Any]], DataFrame],
+    ):
+        super().__init__(query_runner, namespace, server_version)
+        self._handle_properties = handle_properties
+
+
+class GraphPropertyRunner(GraphEntityOpsBaseRunner):
+    @compatible_with("stream", min_inclusive=ServerVersion(2, 2, 0))
+    def stream(self, G: Graph, property: str, element_identifiers: Strings = ["*"], **config: Any) -> DataFrame:
+        self._namespace += ".stream"
+        return self._handle_properties(G, property, element_identifiers, config)
+
+
+class GraphNodePropertiesRunner(GraphEntityOpsBaseRunner):
+    @compatible_with("stream", min_inclusive=ServerVersion(2, 2, 0))
+    def stream(
+        self,
+        G: Graph,
+        node_properties: List[str],
+        node_labels: Strings = ["*"],
+        separate_property_columns: bool = False,
+        **config: Any,
+    ) -> DataFrame:
+        self._namespace += ".stream"
+
+        result = self._handle_properties(G, node_properties, node_labels, config)
+
+        # new format was requested, but the query was run via Cypher
+        if separate_property_columns and "propertyValue" in result.keys():
+            result = result.pivot(index="nodeId", columns="nodeProperty", values="propertyValue")
+            result = result.reset_index()
+            result.columns.name = None
+        # old format was requested but the query was run via Arrow
+        elif not separate_property_columns and "propertyValue" not in result.keys():
+            result = result.melt(id_vars=["nodeId"]).rename(
+                columns={"variable": "nodeProperty", "value": "propertyValue"}
+            )
+
+        return result
+
+    @compatible_with("write", min_inclusive=ServerVersion(2, 2, 0))
+    def write(self, G: Graph, node_properties: List[str], node_labels: Strings = ["*"], **config: Any) -> Series:
```

Comment:
@soerenreichardt 
MyPy complains here `Returning Any from function declared to return "Series"`

Running `./scripts/checkstyle`.

Reply:
similar to previous call-sites we ignore the type warning here, as we know that write/drop will only return a single row